proton-j/src/main/java/org/apache/qpid/proton/engine/impl/EventImpl.java

```diff
@@ -68,12 +69,19 @@ void clear()
     }
 
     @Override
-    public Type getType()
+    public EventType getEventType()
     {
         return type;
     }
 
     @Override
+    public Type getType() {
+        if (type instanceof Type)
```

Comment:
Why this needs to be so is more obvious if you consider how Core events (maybe they should be Engine events) would look if reactor had the ReallyReallyCore events :)

Wdyt, replace Core with Engine in all above comments and leave Core for the deeper refactoring.